pkg/controller/windowsmachineconfig/windows/connectivity.go

```diff
@@ -0,0 +1,135 @@
+package windows
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+// sshPort is the default SSH port
+const sshPort = "22"
+
+// connectivity encapsulates the information needed to connect to the Windows VM over ssh
+type connectivity struct {
+	// username is the user to connect to the VM
+	username string
+	// ipAddress is the VM's IP address
+	ipAddress string
+	// signer is used for authenticating against the VM
+	signer ssh.Signer
+	// sshClient is the client used to access the Windows VM via ssh
+	sshClient *ssh.Client
+}
+
+// newConnectivity returns an instance of connectivity
+func newConnectivity(username, ipAddress string, signer ssh.Signer) (*connectivity, error) {
+	c := &connectivity{
+		username:  username,
+		ipAddress: ipAddress,
+		signer:    signer,
+	}
+	err := c.initSSHClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "error instantiating SSH client")
+	}
+	return c, nil
+}
+
+// initSSHClient returns a key based SSH client
+func (c *connectivity) initSSHClient() error {
+	if c.username == "" || c.ipAddress == "" || c.signer == nil {
+		return fmt.Errorf("incomplete connectivity information: %v", c)
+	}
+
+	config := &ssh.ClientConfig{
+		User: c.username,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(c.signer),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+	sshClient, err := ssh.Dial("tcp", c.ipAddress+":"+sshPort, config)
+	if err != nil {
+		return errors.Wrapf(err, "unable to connect to Windows VM %s", c.ipAddress)
+	}
+	c.sshClient = sshClient
+	return nil
+}
+
+// run instantiates a new SSH session and runs the command on the VM and returns the combined stdout and stderr output
+func (c *connectivity) run(cmd string) (string, error) {
```

Comment:
To me `run`, `transfer` are verbs that should be associated with an interface, WindowsVM in this case, the `connectivity` struct should be used as a config store and windows struct should have these methods not the connectivity struct

Reply:
> I am fine making run, transfer etc part of an interface and a struct like ssh implement them. WDYT?

I am fine with this if you think we're leaking SSH specific things into windows struct. However please note that it will be overkill considering we're not sure if other connectivity methods like WinRM etc can implement these methods.